pkg/service/targetgroupsvc/service.go

```diff
@@ -0,0 +1,25 @@
+package targetgroupsvc
+
+import (
+	"context"
+
+	registry_types "github.com/common-fate/provider-registry-sdk-go/pkg/providerregistrysdk"
+
+	"github.com/benbjohnson/clock"
+	"github.com/common-fate/common-fate/pkg/targetgroup"
+	"github.com/common-fate/common-fate/pkg/types"
+	"github.com/common-fate/ddb"
+)
+
+type Service struct {
+	Clock                  clock.Clock
+	DB                     ddb.Storage
+	ProviderRegistryClient registry_types.ClientWithResponsesInterface
+}
+
+//go:generate go run github.com/golang/mock/mockgen -destination=mocks/targetgroup.go -package=mocks . TargetGroupService
+type TargetGroupService interface {
```

Comment:
In Go and as our own best practice, we choose to add interfaces in the consumer or the interface not in the implementation.

What this means is that if this interface and the generated mocks are used by the API service, then this interface should be defined in the API package instead